contracts/SushiYieldSource.sol

```diff
@@ -1,29 +1,36 @@
 // SPDX-License-Identifier: GPL-3.0
 
 pragma solidity 0.6.12;
-
 import { IYieldSource } from "@pooltogether/yield-source-interface/contracts/IYieldSource.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import "@openzeppelin/contracts/math/SafeMath.sol";
 
 import "./ISushiBar.sol";
-import "./ISushi.sol";
 
 /// @title A pooltogether yield source for sushi token
 /// @author Steffel Fenix
 contract SushiYieldSource is IYieldSource {
-    
+    using SafeERC20 for IERC20;
     using SafeMath for uint256;
     
     ISushiBar public immutable sushiBar;
-    ISushi public immutable sushiAddr;
+    IERC20 public immutable sushiAddr;
     
     mapping(address => uint256) public balances;
 
-    constructor(ISushiBar _sushiBar, ISushi _sushiAddr) public {
+    constructor(ISushiBar _sushiBar, IERC20 _sushiAddr) public {
         sushiBar = _sushiBar;
         sushiAddr = _sushiAddr;
```

Comment:
You should approve max here in the constructor as well, so that the yield source is ready

Reply:
Done: https://github.com/pooltogether/sushi-pooltogether/pull/17/commits/40953c844322a70b4cad8990848cf78085d78465